File: exercism/javascript/raindrops/raindrops.js
const range = require('lodash');

const sounds = {
    3: 'Pling',
    5: 'Plang',
    7: 'Plong'
};
export const convert = (number) => {
    let range_arr = range.range(1, number + 1);

    let factors = range_arr.filter((num) => number % num === 0 && num in sounds);
    let words = factors.map((num) => sounds[num]);

    if (words.length === 0) return number.toString();
    return words.join('');
};

File: exercism/python/raindrops/raindrops.py
def convert(number):
    factors = get_factors(number)
    raindrop_sounds = get_raindrop_sounds(factors)

    if not raindrop_sounds:
        return str(number)
    return raindrop_sounds


def get_factors(number):
    return [i for i in range(1, number + 1) if number % i == 0]


def get_raindrop_sounds(factors):
    sounds = {3: "Pling", 5: "Plang", 7: "Plong"}
    return "".join([sounds[factor] for factor in factors if factor in sounds])
